var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -101,6 +101,8 @@ class Petsc(Package):
               when='@3.7.5%clang@8.1.0:')
     patch('pkg-config-3.7.6-3.8.4.diff', when='@3.7.6:3.8.4')
 
+    patch('xcode_stub_out_of_sync.patch')
```

Comment:
not sure if we should limit this to specific versions

Reply:
Will have this fix in petsc-3.10.5+ so its best to limit this patch to 3.10.4 and lower